tests/handlers/analyze_test.py

```diff
@@ -7,59 +7,57 @@
-from unittest.mock import ANY, call, patch
+from unittest.mock import ANY
 
 import jwt
 
 from jwt_authorizer.config import ALGORITHM
 from jwt_authorizer.session import Ticket
-from jwt_authorizer.tokens import issue_token
-from tests.util import RSAKeyPair, create_test_app
+from tests.util import RSAKeyPair, create_test_app, create_test_token
 
 if TYPE_CHECKING:
-    import redis
+    from aiohttp.pytest_plugin.test_utils import TestClient
 
 
-def test_analyze_ticket(redis_client: redis.Redis) -> None:
-    payload = {
-        "aud": "https://test.example.com/",
-        "email": "some-user@example.com",
-        "iss": "https://orig.example.com/",
-        "jti": "some-unique-id",
-        "sub": "some-user",
-        "uidNumber": "1000",
-    }
+async def test_analyze_ticket(aiohttp_client: TestClient) -> None:
     ticket = Ticket()
     keypair = RSAKeyPair()
+    ticket = Ticket()
+    ticket_handle = ticket.encode("oauth2_proxy")
+    ticket_b64 = base64.urlsafe_b64encode(ticket_handle.encode()).decode()
+    cookie = f"{ticket_b64}|32132781|blahblah"
+    token = create_test_token(
+        keypair,
+        ["admin"],
+        kid="orig-kid",
+        aud="https://test.example.com/",
+        iss="https://orig.example.com/",
+    )
     session_secret = os.urandom(16)
-    app = create_test_app(keypair, session_secret)
+    app = await create_test_app(keypair, session_secret)
+    client = await aiohttp_client(app)
 
     # To test, we need a valid ticket.  The existing code path that creates
     # one is the code path that reissues a JWT based on one from an external
-    # authentication source.  Run that code path, intercepting the call that
-    # attempts to retrieve the public key from a remote server (while checking
-    # that it was called correctly).
-    #
-    # Then, post the resulting ticket to the /analyze endpoint.
-    with app.app_context():
-        issue_token(
-            payload, "https://example.com/", False, ticket, redis_client
-        )
-        with patch("jwt_authorizer.tokens.get_key_as_pem") as get_key_as_pem:
-            get_key_as_pem.return_value = keypair.public_key_as_pem()
-            with app.test_client() as client:
-                response = client.post(
-                    "/auth/analyze",
-                    data={"token": ticket.encode("oauth2_proxy")},
-                )
-            assert get_key_as_pem.call_args_list == [
-                call("https://test.example.com/", "some-kid")
-            ]
+    # authentication source.  Run that code path.
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Bearer {token}"},
+        cookies={"oauth2_proxy": cookie},
+    )
+    assert r.status == 200
+    token = r.headers["X-Auth-Request-Token"]
+
+    # Now pass that token to the /auth/analyze endpoint.
+    r = await client.post(
+        "/auth/analyze", data={"token": ticket.encode("oauth2_proxy")},
```

Comment:
Is this supposed to be token instead of ticket?  A bit confused by by the comment.

Reply:
Ticket, thanks. Fixed the comment.